### PR TITLE
Add Rust Stub

### PR DESCRIPTION
**Pull request by Smiddie31**
# What issue is this addressing?
Adding the stub/section for rust development
## What _type_ of issue is this addressing?
enhancement
## What this PR does | solves
This PR adds the stub/section for the rust language. I have followed the guidelines of adding the new code and contributor expectations. Inside the zone I have added:
:spades: .gitignore
:hearts: devlog.md
:clubs: LICENSE file
:diamonds: Makefile (This is a placeholder and contains examples)
:spades: README file.

### CODE REVIEW

File: rust/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# [Collatz](https://github.com/Skenvy/Collatz): [Rust](https://github.com/Skenvy/Collatz/tree/main/rust) ♥️🦀♥️
+Functions related to [the Collatz/Syracuse/3N+1 problem](https://en.wikipedia.org/wiki/Collatz_conjecture), implemented in [Rust](https://doc.rust-lang.org/reference/index.html).
```
Comment (Skenvy) [inline rust/README.md:2]: Can the link for Rust be https://www.rust-lang.org/ 
I'd probably mention the reference docs in the Devlog though!

File: rust/devlog.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Devlog
```
Comment (Skenvy) [inline rust/devlog.md:1]: Drop a link in here to the 100,000'th issue on rust that the emoji choice came from!

**Review by Skenvy — CHANGES_REQUESTED** [review]
Can we rename the folder (and links to the folder) to "Rust" instead of "rust"

**Comment by Skenvy** [comment]
Add to the main readme at the end of the table
```
| [Rust](https://github.com/Skenvy/Collatz/tree/main/rust) | **#TODO** | **#TODO** | _N/A_ |
```
And to the main devlog
```
* [Rust](https://github.com/Skenvy/Collatz/blob/main/rust/devlog.md)
```

File: rust/.gitignore [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Generated by Cargo
```
[the remark this answers is not in the document]
Comment (Smiddie31) [reply rust/.gitignore:4]: Addressed this by adding the url

File: rust/README.md (hunk 1 above)
Comment (Smiddie31) [reply rust/README.md]: Updated this accordingly to reflect the https://www.rust-lang.org/ url

File: rust/devlog.md (hunk 2 above)
Comment (Smiddie31) [reply rust/devlog.md]: Added the link.

**Review by Skenvy — APPROVED** [review]
Nice, thanks!

**Pull request merged**